src/server/project.js

```diff
@@ -192,9 +193,18 @@ Project.prototype.prepare = function () {
                 currentProjectState = currentState;
 
                 if (this._shouldPrepare(currentProjectState)) {
-                    return prepareUtil.execCordovaPrepare(this.projectRoot, this._lastPlatform)
+                    var platform = this._lastPlatform;
+                    return prepareUtil.execCordovaPrepare(this.projectRoot, platform)
                         .then(function () {
                             return true;
+                        }).catch(function (err) {
+                            err = err.message || err.toString();
+                            var pos = err.toLowerCase().indexOf('error: ');
+                            if (pos > -1) {
+                                err = err.slice(pos + 7);
+                            }
+                            err = err.split('\n')[0];
+                            log.warning('Preparing platform \'' + platform + '\' failed: ' + err);
```

Comment:
Actually this error needs to return something. Right now it's returning void, while the success case is returning true. Is falsy enough here?